File: app/src/androidTest/java/com/example/app/HomeScreenUiTest.java
import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.espresso.action.ViewActions;

import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.isFocusable;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.junit.Assert.assertThat;
import static org.hamcrest.Matchers.greaterThanOrEqualTo;

@RunWith(AndroidJUnit4.class)
public class HomeScreenUiTest {

    @Test
    public void test_dinosaur_mascot_illustration_displayed() {
        ActivityScenario.launch(HomeActivity.class);
        onView(withId(R.id.dino_mascot)).check(matches(isDisplayed()));
    }

    @Test
    public void test_button_height_minimum_64dp() {
        ActivityScenario.launch(HomeActivity.class);
        onView(withText("¡Jugar!")).check((view, e) -> {
            int heightInDp = Math.round(view.getHeight() / view.getResources().getDisplayMetrics().density);
            assertThat(heightInDp, greaterThanOrEqualTo(64));
        });
    }

    @Test
    public void test_keyboard_navigation_focusable() {
        ActivityScenario.launch(HomeActivity.class);
        onView(withText("¡Jugar!")).check(matches(isFocusable()));
    }
}

File: app/src/androidTest/java/com/example/app/HomeScreenUiTest.kt
package com.example.app

import android.view.View
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.*
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.hamcrest.Description
import org.hamcrest.Matcher
import org.hamcrest.TypeSafeMatcher
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class HomeScreenUiTest {

    @get:Rule
    val activityRule = ActivityScenarioRule(HomeActivity::class.java)

    @Test
    fun test_dinosaur_mascot_illustration_displayed() {
        onView(withId(R.id.dino_mascot)).check(matches(isDisplayed()))
    }

    @Test
    fun test_button_height_minimum_64dp() {
        val minHeightDp = 64
        onView(withText("¡Jugar!")).check(matches(withMinHeight(minHeightDp)))
    }

    @Test
    fun test_keyboard_navigation_focusable() {
        onView(withText("¡Jugar!")).check(matches(isFocusable()))
    }

    private fun withMinHeight(minHeightDp: Int): Matcher<View> {
        return object : TypeSafeMatcher<View>() {
            override fun describeTo(description: Description) {
                description.appendText("with minimum height of $minHeightDp dp")
            }

            override fun matchesSafely(view: View): Boolean {
                val density = view.resources.displayMetrics.density
                val minHeightPx = minHeightDp * density
                return view.height >= minHeightPx
            }
        }
    }

    private fun isFocusable(): Matcher<View> {
        return object : TypeSafeMatcher<View>() {
            override fun describeTo(description: Description) {
                description.appendText("is focusable")
            }

            override fun matchesSafely(view: View): Boolean {
                return view.isFocusable
            }
        }
    }
}
